client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -28,523 +26,383 @@ import {
-import { setPayment } from 'lib/upgrades/actions';
-import {
-	BEFORE_SUBMIT,
-	INPUT_VALIDATION,
-	RECEIVED_PAYMENT_KEY_RESPONSE,
-	RECEIVED_WPCOM_RESPONSE,
-	REDIRECTING_FOR_AUTHORIZATION,
-	SUBMITTING_PAYMENT_KEY_REQUEST,
-	SUBMITTING_WPCOM_REQUEST,
-} from 'lib/store-transactions/step-types';
-import notices from 'notices';
+import { webPayment } from 'lib/store-transactions';
+import { setPayment, setStripeObject } from 'lib/upgrades/actions';
+import { setTaxCountryCode, setTaxPostalCode } from 'lib/upgrades/actions/cart';
 import CartToggle from './cart-toggle';
 import CheckoutTerms from './checkout-terms';
 import PaymentChatButton from './payment-chat-button';
 import RecentRenewals from './recent-renewals';
 import SubscriptionText from './subscription-text';
-import { setTaxCountryCode, setTaxPostalCode } from 'lib/upgrades/actions/cart';
+import { withStripe } from 'lib/stripe';
+import { useDebounce } from 'blocks/credit-card-form/helpers';
 
-const debug = debugFactory( 'calypso:checkout:payment:apple-pay' );
+const debug = debugFactory( 'calypso:checkout:payment:web-payment-box' );
 
-/**
- * Supported card types.
- */
-const SUPPORTED_NETWORKS = [ 'visa', 'masterCard', 'discover', 'amex', 'jcb', 'chinaUnionPay' ];
-const APPLE_PAY_MERCHANT_IDENTIFIER = config( 'apple_pay_merchant_id' );
 const PAYMENT_REQUEST_OPTIONS = {
 	requestPayerName: true,
 	requestPayerPhone: false,
 	requestPayerEmail: false,
 	requestShipping: false,
 };
 
-/**
- * The `<WebPaymentBox />` component.
- */
-export class WebPaymentBox extends React.Component {
-	static propTypes = {
-		cart: PropTypes.shape( {
-			currency: PropTypes.string.isRequired,
-			total_cost: PropTypes.number.isRequired,
-			products: PropTypes.array.isRequired,
-		} ).isRequired,
-		transaction: PropTypes.shape( {
-			payment: PropTypes.object,
-		} ).isRequired,
-		transactionStep: PropTypes.shape( {
-			name: PropTypes.string.isRequired,
-			error: PropTypes.object,
-		} ).isRequired,
-		countriesList: PropTypes.array.isRequired,
-		onSubmit: PropTypes.func.isRequired,
-		translate: PropTypes.func.isRequired,
-	};
-
-	state = { processorCountry: 'US' };
-
-	constructor() {
-		super();
-		this.detectedPaymentMethod = detectWebPaymentMethod();
-	}
-
-	/**
-	 * @return {object} A dictionary containing `default`, `disabled` and `text` keys.
-	 */
-	getButtonState = () => {
-		const { transactionStep, translate } = this.props;
-
-		const defaultState = () => {
-			return {
-				default: true,
-				disabled: false,
-				text: translate( 'Select a payment card', {
-					context: 'Loading state on /checkout',
-				} ),
-			};
-		};
-		const ongoingState = () => {
-			return {
-				default: false,
-				disabled: true,
-				text: translate( 'Sending your purchase', {
-					context: 'Loading state on /checkout',
-				} ),
-			};
-		};
-		const completingState = () => {
-			return {
-				default: false,
-				disabled: true,
-				text: translate( 'Completing your purchase', {
```

Comment:
Is there a way to keep these button states in place (or alternatively replace them with some other kind of progress indicator)?

The experience seems pretty bad without them, since once you complete the Apple Pay payment it just closes the payment sheet and shows you the page underneath, which still has the "Buy with Apple Pay" button showing and no indicator that your payment is processing.  Especially for domain purchases, where it can take a while to process the payment, this makes it look like nothing is happening.

Reply:
Good point. I forgot to handle the "processing" state. I think I can do it without this object though, which I felt was an unnecessary abstraction.

Reply:
Done.